lib/cim/cim-adapter.ts

```diff
@@ -0,0 +1,49 @@
+import {PimClass} from "../pim/model";
+import {CoreResourceReader} from "../core";
+import {IriProvider} from "./iri-provider";
+
+export interface CimAdapter {
+
+    /**
+     * Resources returned by the adapter are on the PIM level, therefore the PIM IRIs are needed to interconnect them.
+     * Also, the IRIs need to be predictable and revertible so the user is able to get CIM iri from PIM iri.
+     * @param iriProvider
+     */
+    setIriProvider(iriProvider: IriProvider): void;
+
+    /**
+     * Searches the CIM and returns an ordered list of PimClasses.
+     * @param searchQuery Search query
```

Comment:
Search query is a search query .. yet I do not know the language of the query nor how does the function performs the search, is there some type of label matching for example?